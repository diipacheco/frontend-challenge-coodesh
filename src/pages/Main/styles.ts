import styled from 'styled-components';

export const Container = styled.main`
  display: flex;
  height: 100%;

  flex-direction: column;
`;

export const Content = styled.section`
  display: flex;
  flex-direction: column;

  width: 100%;
  max-width: 650px;

  align-self: center;
  justify-self: center;

  margin-top: 80px;

  > h5 {
    font-size: 16px;
  }
`;
