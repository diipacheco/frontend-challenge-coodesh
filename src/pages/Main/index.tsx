import React, { useEffect } from 'react';

import { useUsers } from '../../hooks/users';

import Header from '../../components/Header';
import SearchBar from '../../components/SearchBar';
import Table from '../../components/Table';

import { Container, Content } from './styles';

const Main: React.FC = () => {
  const { fetchUsers } = useUsers();

  useEffect(() => {
    fetchUsers();
  }, []);
  return (
    <Container>
      <Header />
      <Content>
        <h5>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas
          mollis, magna ac varius fringilla, metus tellus ornare dui, non
          aliquet augue mi auctor massa mauris.
        </h5>

        <SearchBar />

        <Table />
      </Content>
    </Container>
  );
};

export default Main;
