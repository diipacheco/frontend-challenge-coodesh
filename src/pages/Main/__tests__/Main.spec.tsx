import { render } from '@testing-library/react';

import Main from '../index';

describe('<Main />', () => {
  it('should match snapshots', () => {
    const wrapper = render(<Main />);

    expect(wrapper).toMatchSnapshot();
  });
});
