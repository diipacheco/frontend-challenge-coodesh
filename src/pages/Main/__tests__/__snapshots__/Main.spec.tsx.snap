// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshots 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="sc-jrAFXE hZUTjv"
      >
        <header
          class="sc-bdfBQB hKKSWb"
        >
          <h1>
            Pharma Inc
          </h1>
          <div
            class="sc-gsTEea eVigpr"
          >
            <svg
              color="#A2B0BD"
              fill="currentColor"
              height="24"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(162, 176, 189);"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </header>
        <section
          class="sc-kEjbQP kyWxxo"
        >
          <h5>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas mollis, magna ac varius fringilla, metus tellus ornare dui, non aliquet augue mi auctor massa mauris.
          </h5>
          <div
            class="sc-dlfnuX krEQxh"
          >
            <input
              placeholder="Searching"
              type="text"
            />
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 14v2a6 6 0 0 0-6 6H4a8 8 0 0 1 8-8zm0-1c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm9.446 9.032l1.504 1.504-1.414 1.414-1.504-1.504a4 4 0 1 1 1.414-1.414zM18 20a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
                />
              </g>
            </svg>
          </div>
          <div
            class="sc-iBPTik ekuUPc"
          >
            <table>
              <tbody>
                <tr
                  data-testid="categories-row"
                >
                  <th>
                    Name
                    <button
                      data-testid="sort-button"
                      type="button"
                    >
                      <svg
                        fill="#293845"
                        height="20"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 1024 1024"
                        width="20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M839.6 433.8L749 150.5a9.24 9.24 0 0 0-8.9-6.5h-77.4c-4.1 0-7.6 2.6-8.9 6.5l-91.3 283.3c-.3.9-.5 1.9-.5 2.9 0 5.1 4.2 9.3 9.3 9.3h56.4c4.2 0 7.8-2.8 9-6.8l17.5-61.6h89l17.3 61.5c1.1 4 4.8 6.8 9 6.8h61.2c1 0 1.9-.1 2.8-.4 2.4-.8 4.3-2.4 5.5-4.6 1.1-2.2 1.3-4.7.6-7.1zM663.3 325.5l32.8-116.9h6.3l32.1 116.9h-71.2zm143.5 492.9H677.2v-.4l132.6-188.9c1.1-1.6 1.7-3.4 1.7-5.4v-36.4c0-5.1-4.2-9.3-9.3-9.3h-204c-5.1 0-9.3 4.2-9.3 9.3v43c0 5.1 4.2 9.3 9.3 9.3h122.6v.4L587.7 828.9a9.35 9.35 0 0 0-1.7 5.4v36.4c0 5.1 4.2 9.3 9.3 9.3h211.4c5.1 0 9.3-4.2 9.3-9.3v-43a9.2 9.2 0 0 0-9.2-9.3zM310.3 167.1a8 8 0 0 0-12.6 0L185.7 309c-4.2 5.3-.4 13 6.3 13h76v530c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V322h76c6.7 0 10.5-7.8 6.3-13l-112-141.9z"
                        />
                      </svg>
                    </button>
                  </th>
                  <th>
                    Gender
                  </th>
                  <th>
                    Birth
                  </th>
                  <th>
                    Actions
                  </th>
                </tr>
                <tr
                  data-testid="content-row"
                >
                  <td>
                    Sébastien Leroy
                  </td>
                  <td>
                    male
                  </td>
                  <td>
                    03/11/1973
                  </td>
                  <td>
                    <button
                      class="sc-pGacB gkFoUy"
                      type="button"
                    >
                      View
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
            <button
              class="sc-fubCzh erTMDO"
              type="button"
            >
              <svg
                fill="#293845"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 0 0-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 0 1 655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 0 1 279 755.2a342.16 342.16 0 0 1-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 0 1 109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 0 0 3 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
                />
              </svg>
              Loading more...
            </button>
          </div>
        </section>
      </main>
    </div>
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": <div>
    <main
      class="sc-jrAFXE hZUTjv"
    >
      <header
        class="sc-bdfBQB hKKSWb"
      >
        <h1>
          Pharma Inc
        </h1>
        <div
          class="sc-gsTEea eVigpr"
        >
          <svg
            color="#A2B0BD"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(162, 176, 189);"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </header>
      <section
        class="sc-kEjbQP kyWxxo"
      >
        <h5>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas mollis, magna ac varius fringilla, metus tellus ornare dui, non aliquet augue mi auctor massa mauris.
        </h5>
        <div
          class="sc-dlfnuX krEQxh"
        >
          <input
            placeholder="Searching"
            type="text"
          />
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 14v2a6 6 0 0 0-6 6H4a8 8 0 0 1 8-8zm0-1c-3.315 0-6-2.685-6-6s2.685-6 6-6 6 2.685 6 6-2.685 6-6 6zm0-2c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm9.446 9.032l1.504 1.504-1.414 1.414-1.504-1.504a4 4 0 1 1 1.414-1.414zM18 20a2 2 0 1 0 0-4 2 2 0 0 0 0 4z"
              />
            </g>
          </svg>
        </div>
        <div
          class="sc-iBPTik ekuUPc"
        >
          <table>
            <tbody>
              <tr
                data-testid="categories-row"
              >
                <th>
                  Name
                  <button
                    data-testid="sort-button"
                    type="button"
                  >
                    <svg
                      fill="#293845"
                      height="20"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 1024 1024"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M839.6 433.8L749 150.5a9.24 9.24 0 0 0-8.9-6.5h-77.4c-4.1 0-7.6 2.6-8.9 6.5l-91.3 283.3c-.3.9-.5 1.9-.5 2.9 0 5.1 4.2 9.3 9.3 9.3h56.4c4.2 0 7.8-2.8 9-6.8l17.5-61.6h89l17.3 61.5c1.1 4 4.8 6.8 9 6.8h61.2c1 0 1.9-.1 2.8-.4 2.4-.8 4.3-2.4 5.5-4.6 1.1-2.2 1.3-4.7.6-7.1zM663.3 325.5l32.8-116.9h6.3l32.1 116.9h-71.2zm143.5 492.9H677.2v-.4l132.6-188.9c1.1-1.6 1.7-3.4 1.7-5.4v-36.4c0-5.1-4.2-9.3-9.3-9.3h-204c-5.1 0-9.3 4.2-9.3 9.3v43c0 5.1 4.2 9.3 9.3 9.3h122.6v.4L587.7 828.9a9.35 9.35 0 0 0-1.7 5.4v36.4c0 5.1 4.2 9.3 9.3 9.3h211.4c5.1 0 9.3-4.2 9.3-9.3v-43a9.2 9.2 0 0 0-9.2-9.3zM310.3 167.1a8 8 0 0 0-12.6 0L185.7 309c-4.2 5.3-.4 13 6.3 13h76v530c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V322h76c6.7 0 10.5-7.8 6.3-13l-112-141.9z"
                      />
                    </svg>
                  </button>
                </th>
                <th>
                  Gender
                </th>
                <th>
                  Birth
                </th>
                <th>
                  Actions
                </th>
              </tr>
              <tr
                data-testid="content-row"
              >
                <td>
                  Sébastien Leroy
                </td>
                <td>
                  male
                </td>
                <td>
                  03/11/1973
                </td>
                <td>
                  <button
                    class="sc-pGacB gkFoUy"
                    type="button"
                  >
                    View
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
          <button
            class="sc-fubCzh erTMDO"
            type="button"
          >
            <svg
              fill="#293845"
              height="20"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 0 0-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 0 1 655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 0 1 279 755.2a342.16 342.16 0 0 1-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 0 1 109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 0 0 3 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
              />
            </svg>
            Loading more...
          </button>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
